twoliter/src/lock.rs

```diff
@@ -54,9 +54,17 @@ pub(crate) struct LockedImage {
 
 impl LockedImage {
     pub async fn new(vendor: &Vendor, image: &Image) -> Result<Self> {
+        Self::create("manifest", vendor, image).await
+    }
+
+    pub async fn new_sdk(vendor: &Vendor, image: &Image) -> Result<Self> {
+        Self::create("image", vendor, image).await
+    }
+
+    async fn create(item: &str, vendor: &Vendor, image: &Image) -> Result<Self> {
         let source = format!("{}/{}:v{}", vendor.registry, image.name, image.version);
         let manifest_bytes = docker!(
-            ["manifest", "inspect", source.as_str()],
+            [item, "inspect", source.as_str()],
```

Comment:
We need to emit a more detailed error message, right now we just get "failed to inspect ..." with no additional output.